public/app/features/dashboard-scene/settings/AnnotationsEditView.test.tsx

```diff
@@ -0,0 +1,115 @@
+import { map, of } from 'rxjs';
+
+import { DataQueryRequest, DataSourceApi, LoadingState, PanelData } from '@grafana/data';
+import { SceneDataLayers, SceneGridItem, SceneGridLayout, SceneTimeRange } from '@grafana/scenes';
+
+import { AlertStatesDataLayer } from '../scene/AlertStatesDataLayer';
+import { DashboardAnnotationsDataLayer } from '../scene/DashboardAnnotationsDataLayer';
+import { DashboardScene } from '../scene/DashboardScene';
+import { activateFullSceneTree } from '../utils/test-utils';
+
+import { AnnotationsEditView } from './AnnotationsEditView';
+
+const getDataSourceSrvSpy = jest.fn();
+const runRequestMock = jest.fn().mockImplementation((ds: DataSourceApi, request: DataQueryRequest) => {
+  const result: PanelData = {
+    state: LoadingState.Loading,
+    series: [],
+    timeRange: request.range,
+  };
+
+  return of([]).pipe(
+    map(() => {
+      result.state = LoadingState.Done;
+      result.series = [];
+
+      return result;
+    })
+  );
+});
+
+jest.mock('@grafana/runtime', () => ({
+  ...jest.requireActual('@grafana/runtime'),
+  getDataSourceSrv: () => {
+    getDataSourceSrvSpy();
+  },
+  getRunRequest: () => (ds: DataSourceApi, request: DataQueryRequest) => {
+    return runRequestMock(ds, request);
+  },
+  config: {
+    publicDashboardAccessToken: 'ac123',
+  },
+}));
+
+describe('AnnotationsEditView', () => {
+  describe('Dashboard annotations state', () => {
```

Comment:
I think it would be a good idea to add an assertion for each public method since they can be used from anywhere where the instance is created. WDYT?

Reply:
Agreed. Added tests for the other methods as well.